### PR TITLE
Put some unsolvable tests into methods filter

### PR DESCRIPTION
**Pull request by Linary**
Change-Id: Id853ede1bf8277ba2b942e3c502257e78ec23617

### CODE REVIEW

File: hugegraph-test/src/main/java/com/baidu/hugegraph/tinkerpop/ProcessBasicSuite.java [hunk 1]
```diff
@@ -163,7 +163,8 @@
             VertexTest.Traversals.class,
             UnfoldTest.Traversals.class,
             ValueMapTest.Traversals.class,
-            WriteTest.Traversals.class,
+            // Override WriteTest.Traversals.class,
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/tinkerpop/ProcessBasicSuite.java]: remove ","

**Comment by javeme** [comment]
RocksDB CI error: `too many open files`, refer to rocksdb/issues/5488

**Comment by javeme** [comment]
Cassandra CI error: `Batch too large`
image

Cassandra CI error: `Undefined edge label with id: '9238'`
image

**Comment by javeme** [comment]
```
*** Error in `/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java': corrupted size vs. prev_size: 0x00007fa73a7da6d0 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777e5)[0x7fa7b31a87e5]
/lib/x86_64-linux-gnu/libc.so.6(+0x7e9dc)[0x7fa7b31af9dc]
/lib/x86_64-linux-gnu/libc.so.6(+0x81cde)[0x7fa7b31b2cde]
/lib/x86_64-linux-gnu/libc.so.6(__libc_malloc+0x54)[0x7fa7b31b5184]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(_Znwm+0x18)[0x7fa7b18b1e78]
/tmp/librocksdbjni5858602310051612369.so(_ZN7rocksdb25NewArenaWrappedDbIteratorEPNS_3EnvERKNS_11ReadOptionsERKNS_18ImmutableCFOptionsERKNS_16MutableCFOptionsERKmmmPNS_12ReadCallbackEPNS_6DBImplEPNS_16ColumnFamilyDataEbb+0x39)[0x7fa7675ed2f9]
/tmp/librocksdbjni5858602310051612369.so(_ZN7rocksdb6DBImpl15NewIteratorImplERKNS_11ReadOptionsEPNS_16ColumnFamilyDataEmPNS_12ReadCallbackEbb+0x84)[0x7fa767578db4]
/tmp/librocksdbjni5858602310051612369.so(_ZN7rocksdb6DBImpl11NewIteratorERKNS_11ReadOptionsEPNS_18ColumnFamilyHandleE+0x94)[0x7fa767578ed4]
/tmp/librocksdbjni5858602310051612369.so(Java_org_rocksdb_RocksDB_iteratorCF__JJJ+0xc7)[0x7fa7674e6127]
[0x7fa79d77aa2e]
======= Memory map: ========
00400000-00401000 r-xp 00000000 08:01 543321                             /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
00600000-00601000 r--p 00000000 08:01 543321                             /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
00601000-00602000 rw-p 00001000 08:01 543321                             /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
01e78000-03eb6000 rw-p 00000000 00:00 0                                  [heap]
8b200000-90f80000 rw-p 00000000 00:00 0 
90f80000-d9100000 ---p 00000000 00:00 0 
d9100000-f7900000 rw-p 00000000 00:00 0 
f7900000-100000000 ---p 00000000 00:00 0 
100000000-1003e0000 rw-p 00000000 00:00 0 
1003e0000-140000000 ---p 00000000 00:00 0 
7fa72c000000-7fa72f001000 rw-p 00000000 00:00 0 
7fa72f001000-7fa730000000 ---p 00000000 00:00 0 
7fa730000000-7fa733031000 rw-p 00000000 00:00 0 
7fa733031000-7fa734000000 ---p 00000000 00:00 0 
7fa734000000-7fa737001000 rw-p 00000000 00:00 0 
7fa737001000-7fa738000000 ---p 00000000 00:00 0 
7fa738000000-7fa73b001000 rw-p 00000000 00:00 0 
7fa73b001000-7fa73c000000 ---p 00000000 00:00 0 
7fa73c000000-7fa73f001000 rw-p 00000000 00:00 0 
7fa73f001000-7fa740000000 ---p 00000000 00:00 0 
7fa740000000-7fa743001000 rw-p 00000000 00:00 0 
7fa743001000-7fa744000000 ---p 00000000 00:00 0 
7fa744000000-7fa747001000 rw-p 00000000 00:00 0 
7fa747001000-7fa748000000 ---p 00000000 00:00 0 
7fa74affe000-7fa74afff000 ---p 00000000 00:00 0 
7fa74afff000-7fa74b7ff000 rw-p 00000000 00:00 0 
7fa74b7ff000-7fa74b800000 ---p 00000000 00:00 0 
7fa74b800000-7fa74c000000 rw-p 00000000 00:00 0 
7fa74c000000-7fa74e003000 rw-p 00000000 00:00 0 
7fa74e003000-7fa750000000 ---p 00000000 00:00 0 
7fa7507f9000-7fa7507fa000 ---p 00000000 00:00 0 
7fa7507fa000-7fa750ffa000 rw-p 00000000 00:00 0 
7fa7517fb000-7fa7517fc000 ---p 00000000 00:00 0 
7fa7517fc000-7fa751ffc000 rw-p 00000000 00:00 0 
7fa7527fd000-7fa7527fe000 ---p 00000000 00:00 0 
7fa7527fe000-7fa752ffe000 rw-p 00000000 00:00 0 
7fa752ffe000-7fa752fff000 ---p 00000000 00:00 0 
7fa752fff000-7fa7537ff000 rw-p 00000000 00:00 0 
7fa7537ff000-7fa753800000 ---p 00000000 00:00 0 
7fa753800000-7fa754000000 rw-p 00000000 00:00 0 
7fa754000000-7fa756006000 rw-p 00000000 00:00 0 
7fa756006000-7fa758000000 ---p 00000000 00:00 0 
7fa7587f1000-7fa7587f2000 ---p 00000000 00:00 0 
7fa7587f2000-7fa758ff2000 rw-p 00000000 00:00 0 
7fa758ff2000-7fa758ff3000 ---p 00000000 00:00 0 
7fa758ff3000-7fa7597f3000 rw-p 00000000 00:00 0 
7fa7597f3000-7fa7597f4000 ---p 00000000 00:00 0 
7fa7597f4000-7fa759ff4000 rw-p 00000000 00:00 0 
7fa75a8f6000-7fa75a8f9000 ---p 00000000 00:00 0 
7fa75a8f9000-7fa75a9f7000 rw-p 00000000 00:00 0 
7fa75a9f7000-7fa75a9fa000 ---p 00000000 00:00 0 
7fa75a9fa000-7fa75aaf8000 rw-p 00000000 00:00 0 
7fa75baf9000-7fa75bafc000 ---p 00000000 00:00 0 
7fa75bafc000-7fa75bbfa000 rw-p 00000000 00:00 0 
7fa75bbfa000-7fa75bbfd000 ---p 00000000 00:00 0 
7fa75bbfd000-7fa75bcfb000 rw-p 00000000 00:00 0 
7fa75ebfd000-7fa75ec3d000 rw-p 00000000 00:00 0 
7fa75ec3d000-7fa75ee00000 ---p 00000000 00:00 0 
7fa75ee00000-7fa75eefe000 rw-p 00000000 00:00 0 
7fa75eefe000-7fa75eeff000 ---p 00000000 00:00 0 
7fa75eeff000-7fa75f8ff000 rw-p 00000000 00:00 0 
7fa75f8ff000-7fa75faff000 rw-p 00000000 00:00 0 
7fa75faff000-7fa75fcff000 rw-p 00000000 00:00 0 
7fa75fcff000-7fa75feff000 rw-p 00000000 00:00 0 
7fa75feff000-7fa75ff02000 ---p 00000000 00:00 0 
7fa75ff02000-7fa760000000 rw-p 00000000 00:00 0 
7fa760000000-7fa76200d000 rw-p 00000000 00:00 0 
7fa76200d000-7fa764000000 ---p 00000000 00:00 0 
7fa764027000-7fa764227000 rw-p 00000000 00:00 0 
7fa764227000-7fa76422a000 ---p 00000000 00:00 0 
7fa76422a000-7fa764328000 rw-p 00000000 00:00 0 
7fa764328000-7fa764329000 ---p 00000000 00:00 0 
7fa764329000-7fa764b29000 rw-p 00000000 00:00 0 
7fa764b29000-7fa764b2a000 ---p 00000000 00:00 0 
7fa764b2a000-7fa76532a000 rw-p 00000000 00:00 0 
7fa76532a000-7fa76532b000 ---p 00000000 00:00 0 
7fa76532b000-7fa765b2b000 rw-p 00000000 00:00 0 
7fa765b2b000-7fa765b2c000 ---p 00000000 00:00 0 
7fa765b2c000-7fa76632c000 rw-p 00000000 00:00 0 
7fa76632c000-7fa76632d000 ---p 00000000 00:00 0 
7fa76632d000-7fa766b2d000 rw-p 00000000 00:00 0 
7fa766b2d000-7fa766b2e000 ---p 00000000 00:00 0 
7fa766b2e000-7fa76732e000 rw-p 00000000 00:00 0 
7fa76732e000-7fa7679de000 r-xp 00000000 08:01 1145                       /tmp/librocksdbjni5858602310051612369.so
7fa7679de000-7fa767bde000 ---p 006b0000 08:01 1145                       /tmp/librocksdbjni5858602310051612369.so
7fa767bde000-7fa767bf0000 r--p 006b0000 08:01 1145                       /tmp/librocksdbjni5858602310051612369.so
7fa767bf0000-7fa767bf9000 rw-p 006c2000 08:01 1145                       /tmp/librocksdbjni5858602310051612369.so
7fa767bf9000-7fa767e00000 rw-p 00000000 00:00 0 
```

**Comment by Linary** [comment]
rocksdb version: 6.3.6

```java
public class TestIteratorAfterCloseCF {

    public static void main(String args[]) throws Exception {
        RocksDB rocksdb = testReleaseIteratorAfterCloseCF();

        println("gc for Iterator before");
        System.gc(); Thread.sleep(1000 * 2);
        println("gc for Iterator after");

        System.gc(); Thread.sleep(1000 * 2);
        println("gc2 for Iterator after");

        System.gc(); Thread.sleep(1000 * 2);
        println("gc3 for Iterator after");

        println("close rocksdb");
        rocksdb.close();

        System.gc(); Thread.sleep(1000 * 2);
        println("done");
    }

    public static RocksDB testReleaseIteratorAfterCloseCF() throws Exception {
        String path = "rocksdb-data";
        FileUtils.deleteDirectory(new File(path));

        DBOptions options = new DBOptions();
        options.setWalDir(path);
        options.setCreateIfMissing(true);

        List<ColumnFamilyDescriptor> cfds = new ArrayList<>();
        cfds.add(new ColumnFamilyDescriptor("default".getBytes()));
        List<ColumnFamilyHandle> cfhs = new ArrayList<>();
        RocksDB rocksdb = RocksDB.open(options, path, cfds, cfhs);

        ColumnFamilyDescriptor cfd = new ColumnFamilyDescriptor("cf".getBytes());
        println("create CF");
        ColumnFamilyHandle cfHandle = rocksdb.createColumnFamily(cfd);

        println("new Iterator");
        RocksIterator iter = rocksdb.newIterator(cfHandle);
        iter.seekToFirst();
//        iter.close(); // it's no coredump if manually closed here

        println("drop CF");
        rocksdb.dropColumnFamily(cfHandle); // effect when close
        println("close CF");
        cfHandle.close(); // <== assertion failed: (is_last_reference) here in v6.3.6, it's ok in v5.14.2
        println("closed CF");

        iter.close();

        cfhs.clear(); System.gc(); Thread.sleep(1000);

        return rocksdb;
    }

    private static void println(String line) {
        System.out.println(line);
        System.out.flush();
    }
}
```

```
Stack: [0x000070000a8cc000,0x000070000a9cc000],  sp=0x000070000a9cb500,  free space=1021k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [librocksdbjni5153628548550132452.jnilib+0x12f28e]  rocksdb::MemTableListVersion::Unref(rocksdb::autovector<rocksdb::MemTable*, 8ul>*)+0xe
C  [librocksdbjni5153628548550132452.jnilib+0x596ad]  rocksdb::SuperVersion::Cleanup()+0x1d
C  [librocksdbjni5153628548550132452.jnilib+0x99fb4]  rocksdb::DBImpl::BackgroundCallPurge()+0x304
C  [librocksdbjni5153628548550132452.jnilib+0x25b2ec]  rocksdb::Cleanable::~Cleanable()+0x1c
C  [librocksdbjni5153628548550132452.jnilib+0x101890]  rocksdb::NewArenaWrappedDbIterator(rocksdb::Env*, rocksdb::ReadOptions const&, rocksdb::ImmutableCFOptions const&, rocksdb::MutableCFOptions const&, unsigned long long const&, unsigned long long, unsigned long long, rocksdb::ReadCallback*, rocksdb::DBImpl*, rocksdb::ColumnFamilyData*, bool, bool)+0x4330
C  [librocksdbjni5153628548550132452.jnilib+0xfd0ec]  rocksdb::ArenaWrappedDBIter::~ArenaWrappedDBIter()+0x1c
j  org.rocksdb.RocksIterator.disposeInternal(J)V+0
j  org.rocksdb.AbstractRocksIterator.disposeInternal()V+15
j  org.rocksdb.AbstractImmutableNativeReference.close()V+13
j  com.baidu.hugegraph.example.TestIteratorAfterCloseCF.testReleaseIteratorAfterCloseCF()Lorg/rocksdb/RocksDB;+154
j  com.baidu.hugegraph.example.TestIteratorAfterCloseCF.main([Ljava/lang/String;)V+0
v  ~StubRoutines::call_stub
V  [libjvm.dylib+0x2ed9d6]  JavaCalls::call_helper(JavaValue*, methodHandle*, JavaCallArguments*, Thread*)+0x6ae
V  [libjvm.dylib+0x3247ce]  jni_invoke_static(JNIEnv_*, JavaValue*, _jobject*, JNICallType, _jmethodID*, JNI_ArgumentPusher*, Thread*)+0x1bf
V  [libjvm.dylib+0x31d545]  jni_CallStaticVoidMethod+0x15d
C  [java+0x3931]  JavaMain+0x9c4
C  [libsystem_pthread.dylib+0x5d36]  _pthread_start+0x7d
C  [libsystem_pthread.dylib+0x258f]  thread_start+0xf
C  0x0000000000000000

Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  org.rocksdb.RocksIterator.disposeInternal(J)V+0
j  org.rocksdb.AbstractRocksIterator.disposeInternal()V+15
j  org.rocksdb.AbstractImmutableNativeReference.close()V+13
j  com.baidu.hugegraph.example.TestIteratorAfterCloseCF.testReleaseIteratorAfterCloseCF()Lorg/rocksdb/RocksDB;+154
j  com.baidu.hugegraph.example.TestIteratorAfterCloseCF.main([Ljava/lang/String;)V+0
v  ~StubRoutines::call_stub
```

**Comment by zhoney** [comment]
image

```
com.baidu.hugegraph.HugeException: Failed to update/query TaskStore

	at com.baidu.hugegraph.task.TaskScheduler.call(TaskScheduler.java:419)
	at com.baidu.hugegraph.task.TaskScheduler.call(TaskScheduler.java:411)
	at com.baidu.hugegraph.task.TaskScheduler.close(TaskScheduler.java:219)
	at com.baidu.hugegraph.task.TaskManager.closeScheduler(TaskManager.java:73)
	at com.baidu.hugegraph.HugeGraph.close(HugeGraph.java:494)
	at com.baidu.hugegraph.tinkerpop.TestGraph.close(TestGraph.java:224)
	at com.baidu.hugegraph.tinkerpop.TestGraphProvider.clear(TestGraphProvider.java:327)
	at org.apache.tinkerpop.gremlin.GraphManager$ManagedGraphProvider.clear(GraphManager.java:142)
	at org.apache.tinkerpop.gremlin.structure.GraphConstructionTest.shouldMaintainOriginalConfigurationObjectGivenToFactory(GraphConstructionTest.java:83)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.apache.tinkerpop.gremlin.AbstractGremlinSuite.runChild(AbstractGremlinSuite.java:229)
	at org.apache.tinkerpop.gremlin.AbstractGremlinSuite.runChild(AbstractGremlinSuite.java:51)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at com.baidu.hugegraph.tinkerpop.StructureBasicSuite$1.evaluate(StructureBasicSuite.java:119)
	at org.apache.tinkerpop.gremlin.AbstractGremlinSuite$1.evaluate(AbstractGremlinSuite.java:238)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:237)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: HBase store has not been initialized
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at com.baidu.hugegraph.task.TaskScheduler.call(TaskScheduler.java:417)
	... 45 more
Caused by: java.lang.IllegalStateException: HBase store has not been initialized
	at com.google.common.base.Preconditions.checkState(Preconditions.java:199)
	at com.baidu.hugegraph.util.E.checkState(E.java:68)
	at com.baidu.hugegraph.backend.store.hbase.HbaseStore.checkConnectionOpened(HbaseStore.java:344)
	at com.baidu.hugegraph.backend.store.hbase.HbaseStore.opened(HbaseStore.java:163)
	at com.baidu.hugegraph.backend.store.AbstractBackendStore.checkOpened(AbstractBackendStore.java:57)
	at com.baidu.hugegraph.backend.store.hbase.HbaseStore.close(HbaseStore.java:155)
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.close(AbstractTransaction.java:204)
	at com.baidu.hugegraph.backend.tx.IndexableTransaction.close(IndexableTransaction.java:83)
	at com.baidu.hugegraph.task.TaskScheduler.lambda$close$4(TaskScheduler.java:221)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

```

**Comment by zhoney** [comment]
HBase 错误为sessions计数不对，第一次打开没有+1，修复办法为：

```
diff --git a/hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseStore.java b/hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseStore.java
index 4c3a4ea5..3eeac68a 100644
--- a/hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseStore.java
+++ b/hugegraph-hbase/src/main/java/com/baidu/hugegraph/backend/store/hbase/HbaseStore.java
@@ -147,6 +147,7 @@ public abstract class HbaseStore extends AbstractBackendStore<Session> {
                      "try to init CF later", this.store, this.namespace);
         }
 
+        this.sessions.session();
         LOG.debug("Store opened: {}", this.store);
     }

```

File: hugegraph-test/src/main/java/com/baidu/hugegraph/tinkerpop/StructureBasicSuite.java [hunk 2]
```diff
@@ -81,8 +81,7 @@
             FeatureSupportTest.class,
             IoCustomTest.class,
             IoEdgeTest.class,
-            // TODO: Open it when rocksdb bug fixed
-            // IoGraphTest.class,
+            IoGraphTest.class,
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/tinkerpop/StructureBasicSuite.java]: move to the previous commit

File: .travis.yml [hunk 3]
```diff
@@ -30,11 +30,15 @@ script:
     fi
   - |
     if [[ "$TRAVIS_BRANCH" =~ $RELEASE_BRANCH ]]; then
-      mvn test -P tinkerpop-structure-test,$BACKEND
+      if [[ "$BACKEND" != "rocksdb" && "$BACKEND" != "hbase" ]]; then
+        mvn test -P tinkerpop-structure-test,$BACKEND
```
Comment (javeme) [inline .travis.yml]: add some comments

**Pull request merged**